spec/lib/quickbooks/service/reports_spec.rb

```diff
@@ -1,68 +1,51 @@
-describe "Quickbooks::Service::Reports" do
+describe "Quickbooks::Service::Report" do
   before(:all) do
     construct_service :reports
   end
 
-  it "can query for Balance Sheets", focus: true do
-    xml = fixture("balancesheet.xml")
-    model = Quickbooks::Model::Reports
-
-    stub_request(:get, @service.url_for_query, ["200", "OK"], xml)
-    reports = @service.query('BalanceSheet')
-
-    balance_sheet_xml = @service.last_response_xml
-    balance_sheet_xml.xpath('//xmlns:ReportName').children.to_s == 'BalanceSheet'
+  describe '.url_for_query()' do
+    it 'uses "BalanceSheet" and "This Fiscal Year to date" as the default parameters' do
+      @service.url_for_query().should == 'https://quickbooks.api.intuit.com/v3/company/9991111222/reports/BalanceSheet?date_macro=This+Fiscal+Year-to-date'
+    end
+
+    it 'allows overriding the report type' do
+      @service.url_for_query('ProfitAndLoss').should == 'https://quickbooks.api.intuit.com/v3/company/9991111222/reports/ProfitAndLoss?date_macro=This+Fiscal+Year-to-date'
+    end
+
+    it 'allows overriding the date_macro' do
+      @service.url_for_query('BalanceSheet', 'Last Year').should == 'https://quickbooks.api.intuit.com/v3/company/9991111222/reports/BalanceSheet?date_macro=Last+Year'
+    end
+
+    it 'allows passing additional query parameters' do
+      url = @service.url_for_query('BalanceSheet', 'Last Year',
+        :start_date => '2015-01-01',
+        :end_date => '2015-01-31',
+        :accounting_method => 'Cash',
+        :columns => 'subt_nat_amount,tax_amount',
+      )
+      url.should == 'https://quickbooks.api.intuit.com/v3/company/9991111222/reports/BalanceSheet?start_date=2015-01-01&end_date=2015-01-31&accounting_method=Cash&columns=subt_nat_amount,tax_amount'
+    end
+
+    it 'currently ignores the date_macro argument when passed in additional options' do
```

Comment:
This current behavior seems like a bug to me.  It seems like date_macro should just be another param in the options hash, and not be a positional argument.